src/collectors/jolokia/kafka_jolokia.py

```diff
@@ -0,0 +1,592 @@
+"""
+Collects Kafka JMX metrics from the Jolokia Agent.  Extends the
+JolokiaCollector to reformat the Kafka MBean names into easier to
+understand metric names.
+
+Additionally, we also connect to Zookeeper to collect certain 
+metrics that are critical to Kafka monitoring.  This is done in
+two ways:
+    1) By extending the ProcessCollector, we can execute a command-
+        line function against the Zookeeper instance to extract 
+        infomation about the consumer groups and their corresponding
+        consumer lag.  This replaces and improves upon the 
+        KafkaConsumerLagCollector.
+    2) By extending the ZookeeperCollector, we can collect the 
+        key Zookeeper metrics for Kafka without requiring the user
+        to configure a separate collector. This does not replace
+        the Zookeeper collector, however; users with a Zookeeper
+        instance being used for something other than Kafka can still
+        use the Zookeeper collector on its own. 
+
+Netuitive Change History
+    2016/10/26 DVG - Initial version.
+    2016/11/16 DVG - Added debug logging, additional error handling, refactoring of common code across the Kafka 8 and 9 handling.
+                     Fixed bug with Zookeeper collector defaulting to localhost:2181
+"""
+
+from jolokia import JolokiaCollector
+from zookeeper import ZookeeperCollector
+from diamond.collector import ProcessCollector
```

Comment:
That's a valid point.  And frankly, the idea of Collector A inheriting from Collector B simply so it could reuse come of its code and not because they were actually related was bugging me.  OK, I'll take another pass at this, break those links with the other collectors, and see if I can end up something much cleaner, albeit with some code duplication.